Add pagination support to project assignments API

NKSurprise opened the pull request:
Introduces a pagination infrastructure with abstractions, extensions, and headers. Updates ProjectAssignmentsController and ProjectAssignmentService to support paged retrieval of project assignments. Refactors service and controller interfaces to use paged results, improving scalability and API usability.

NKSurprise committed:
Add pagination support to project assignments API

Introduces a pagination infrastructure with abstractions, extensions, and headers. Updates ProjectAssignmentsController and ProjectAssignmentService to support paged retrieval of project assignments. Refactors service and controller interfaces to use paged results, improving scalability and API usability.

NKSurprise committed:
Increase default page size for project assignments

Changed the default pageSize parameter in GetAll from 2 to 20 to return more project assignments per page.

ImSk1 commented on src/SkillBridge/Infrastructure/Pagination/Headers/PaginationHeadersFilter.cs:
This filter has to be added as an MVC filter for the headers to work correctly.

<img width="742" height="235" alt="image" src="https://github.com/user-attachments/assets/588ba435-d4de-4e91-bd62-718cb556ca21" />

NKSurprise committed:
Add PaginationHeadersFilter to MVC filters

Registered PaginationHeadersFilter in the MVC filter pipeline within ServiceCollectionExtensions. This ensures pagination headers are automatically added to responses where applicable.

ImSk1 committed:
Merge branch 'dev' into 13-implement-pagination-for-api-endpoints-in-skillbridge-api
